src/models/PlanFeedback.ts

```diff
@@ -0,0 +1,121 @@
+  }
+
+  // Validation to be used prior to saving the record
+  async isValid(): Promise<boolean> {
+    await super.isValid();
+
+    if (!this.planId) this.addError('planId', 'Plan can\'t be blank');
+    if (!this.requested) this.addError('requested', 'Requested can\'t be blank');
+    if (!this.requestedById) this.addError('requestedById', 'RequestedById can\'t be blank');
+
+    return Object.keys(this.errors).length === 0;
+  }
+
+  // Ensure data integrity
+  prepForSave(): void {
+    // Remove leading/trailing blank spaces
+    this.summaryText = this.summaryText?.trim();
+  }
+
+  //Create new PlanFeedback
+  async create(context: MyContext): Promise<PlanFeedback> {
+    const reference = 'PlanFeedback.create';
+
+    // First make sure the record is valid
+    if (await this.isValid()) {
+
+      // My assumption is that you can have multiple rounds of requests for the same planId, so we don't need to check if it already exists
+      // Save the record and then fetch it
+      const newId = await PlanFeedback.insert(context, PlanFeedback.tableName, this, reference, ['planFeedbackComments']);
+      const response = await PlanFeedback.findById(reference, context, newId);
+      return response;
+
+    }
+    // Otherwise return as-is with all the errors
+    return new PlanFeedback(this);
+  }
+
+  //Update an existing PlanFeedback
+  async update(context: MyContext, noTouch = false): Promise<PlanFeedback> {
+    if (await this.isValid()) {
+      if (this.id) {
+        await PlanFeedback.update(context, PlanFeedback.tableName, this, 'PlanFeedback.update', ['planFeedbackComments'], noTouch);
+        return await PlanFeedback.findById('PlanFeedback.update', context, this.id);
+      }
+      // This feedback has never been saved before so we cannot update it!
+      this.addError('general', 'PlanFeedback has never been saved');
+    }
+    return new PlanFeedback(this);
+  }
+
+  //Delete the PlanFeedback
+  async delete(context: MyContext): Promise<PlanFeedback> {
+    if (this.id) {
+      const deleted = await PlanFeedback.findById('PlanFeedback.delete', context, this.id);
+
+      const successfullyDeleted = await PlanFeedback.delete(
+        context,
+        PlanFeedback.tableName,
+        this.id,
+        'PlanFeedback.delete'
+      );
+      if (successfullyDeleted) {
+        return deleted;
+      } else {
+        return null
+      }
+    }
+    return null;
+  }
+
+  // Fetch a PlanFeedback by it's id
+  static async findById(reference: string, context: MyContext, feedbackId: number): Promise<PlanFeedback> {
+    const sql = `SELECT * FROM ${PlanFeedback.tableName} WHERE id = ?`;
+    const results = await PlanFeedback.query(context, sql, [feedbackId?.toString()], reference);
+    return Array.isArray(results) && results.length > 0 ? new PlanFeedback(results[0]) : null;
+  }
+
+  // Fetch PlanFeedback by it's planId and requestedById
+  static async findByPlanIdAndRequestedById(
```

Comment:
If the plan allows for multiple rounds of feedback, then a single user can potentially request feedback more than once, so this should return an array instead of a single record.